sdk-wasm-js/src/hash.rs

```diff
@@ -0,0 +1,69 @@
+//! Wrapper over `solana_hash::Hash` with wasm-bindgen
+use {
+    js_sys::{Array, Uint8Array},
+    std::{boxed::Box, format, string::String, vec},
+    wasm_bindgen::{prelude::*, JsCast},
+};
+
+#[wasm_bindgen]
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct Hash(pub(crate) solana_hash::Hash);
+
+#[allow(non_snake_case)]
+#[wasm_bindgen]
+impl Hash {
+    /// Create a new Hash object
+    ///
+    /// * `value` - optional hash as a base58 encoded string, `Uint8Array`, `[number]`
+    #[wasm_bindgen(constructor)]
+    pub fn constructor(value: JsValue) -> Result<Self, JsValue> {
+        if let Some(base58_str) = value.as_string() {
+            base58_str
+                .parse::<solana_hash::Hash>()
+                .map(Hash)
+                .map_err(|x| JsValue::from(x.to_string()))
+        } else if let Some(uint8_array) = value.dyn_ref::<Uint8Array>() {
+            <[u8; solana_hash::HASH_BYTES]>::try_from(uint8_array.to_vec())
+                .map(solana_hash::Hash::new_from_array)
+                .map(Hash)
+                .map_err(|err| format!("Invalid Hash value: {err:?}").into())
+        } else if let Some(array) = value.dyn_ref::<Array>() {
+            let mut bytes = vec![];
+            let iterator = js_sys::try_iter(&array.values())?.expect("array to be iterable");
```

Comment:
I know this is copied over and this probably beyond the scope of this PR, but there are a few unwraps and expects in the code. A panic here can't be caught on the JS side of things and causes an abort which could crash a node/browser process.

Reply:
I see two `unwrap()`s:

* console_log::init_with_level
* bincode::serialize(...)

We can probably fix those, since I could imagine those being fallible.

And the expects are both from `js_sys::try_iter`, after checking the underlying type is an `Array` and passing in `array.values()`, which is an `Iterator`. I think the `expect` there makes more sense because it should likely be a programmer error if the conversion fails.

Since @febo is waiting on this, I'll fix the unwraps in a follow-up